### PR TITLE
Add automated test following waypoints

### PR DESCRIPTION
**Pull request by Tadinu**
* Add under `ExternalTest` dir: automated test `turtlebot3_tests` package
* When the test is deployed with CI, an auto-shutdown Sim command should be called after the test is finished.

### CODE REVIEW

**Review by tetorea — COMMENTED** [review]
I got this when executing `./RunTB3WaypointFollowerTest.sh` :
`./RunTB3WaypointFollowerTest.sh: 5: launch_test: not found`
What is the proper way to build the package?

**Comment by Tadinu** [comment]
@tetorea The launch_testing package is [here](https://github.com/ros2/launch/tree/master/launch_testing), which should be installed default as part of ros2' launch? If not, it could be done by `sudo apt install ros-foxy-launch-testing`, I will update README.

File: ExternalTest/turtlebot3_tests/src/test_waypoint_follower.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
+#! /usr/bin/env python3
+# Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
```
[the remark this answers is not in the document]
Comment (Tadinu) [reply ExternalTest/turtlebot3_tests/src/test_waypoint_follower.py:2]: Yeah, currently all of our sources still have the old one, I suppose when we officially have for 2022 one, we could do that all together? @yuokamoto

[the remark this answers is not in the document]
Comment (yuokamoto) [reply ExternalTest/turtlebot3_tests/src/test_waypoint_follower.py:2]: Can you create different pr to update this?
better to separate in different commit(easier to check)

File: ExternalTest/turtlebot3_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# turtlebot3 tests
+This package contains test cases using the ``launch`` and ``launch_testing`` packages.
+
+## Examples
+
+### `test_waypoint_follower.py`
+
+Usage:
+
+```sh
```
Comment (yuokamoto) [inline ExternalTest/turtlebot3_tests/README.md:10]: I guess this step is just replaced with ./RunTB3WaypointFollowerTest.sh

**Comment by yuokamoto** [comment]
seems this script start gazebo.

File: ExternalTest/turtlebot3_tests/src/test_waypoint_follower.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tadinu) [reply ExternalTest/turtlebot3_tests/src/test_waypoint_follower.py:2]: Yeap, Wrike task added [here](https://www.wrike.com/open.htm?id=845023569), when we have the new precise copyrights note, can you paste it there?

**Comment by Tadinu** [comment]
Right, this test invokes nav2_bringup's `tb3_simulation_launch.py`, which indeed starts with `headless` as False by default [here](https://github.com/ros-planning/navigation2/blob/main/nav2_bringup/launch/tb3_simulation_launch.py#L132). Updated.

**Comment by yuokamoto** [comment]
```
yu@system76-pc:~/turtlebot3-UE/ExternalTest/turtlebot3_tests$ ./RunTB3WaypointFollowerTest.sh 
mkdir: cannot create directory ‘maps’: File exists
ln: failed to create hard link 'maps/Turtlebot3_benchmark.pgm': File exists
ln: failed to create hard link 'maps/Turtlebot3_benchmark.yaml': File exists
Warning: The parameter 'node_executable' is deprecated, use 'executable' instead
Warning: The parameter 'node_executable' is deprecated, use 'executable' instead
Warning: The parameter 'node_name' is deprecated, use 'name' instead
usage: launch_test [-h] [--package-name PACKAGE_NAME] [-v] [-s]
                   [--junit-xml XMLPATH]
                   launch_test_file
                   [launch_arguments [launch_arguments ...]]
launch_test: error: malformed launch argument 'tb3_model:=', expected format '<name>:=<value>'

**Comment by Tadinu** [comment]
Yeap you need to pass it the `<tb3_model>` as burger or waffle, as updated here [31e1356](https://github.com/rapyuta-robotics/turtlebot3-UE/pull/22/commits/31e135642740c683602abc1ad44af3c06967b374)

**Comment by yuokamoto** [comment]
When I executed `./RunTB3WaypointFollowerTest.sh burger`,  rviz started but nothing happen. anyother step required to execute test?

**Comment by Tadinu** [comment]
Yeah that has been observed and indeed does happen occasionally, like even as usual:
* When you manually run the tb3 nav2 command, then if Rviz could not subscribe to the published topics (like `odom, scan`), typically, the steps would be:
-> Restart the Sim.
-> Then only after the map is fully loaded with TB3's odom + scan on, the nav command script should be run.

As for the prerequisite conditions, there should be prior tests that check for Sim state's service server, clock, laser scan, which are essentially required in advance of further functional tests (like nav in this case for Tb3). As such, those tests, actually being added for `io-amr-ue`, should be also added for `tb3-ue` here, but rather in another PR.

File: ExternalTest/turtlebot3_tests/turtlebot3_tests/test_waypoint_follower.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        self.get_logger().warn(msg)
+
+    def error_msg(self, msg: str):
+        self.get_logger().error(msg)
+
+def follow_waypoints():
+    # wait a few seconds to make sure entire stacks are up
+    time.sleep(10)
+
+    wps = [[-0.52, -0.78], [-0.7, 0.5], [2.0, -1.5], [1.7, 1.7]]
```
Comment (yuokamoto) [inline ExternalTest/turtlebot3_tests/turtlebot3_tests/test_waypoint_follower.py:174]: Better to get from launch arg or load from yaml file in the future.

**Review by yuokamoto — APPROVED** [review]

**Pull request merged**